src/java.base/share/classes/java/util/concurrent/ThreadPoolExecutor.java

```diff
@@ -2081,7 +2088,18 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
     /**
      * A handler for rejected tasks that discards the oldest unhandled
      * request and then retries {@code execute}, unless the executor
-     * is shut down, in which case the task is discarded.
+     * is shut down, in which case the task is discarded. This policy is
+     * rarely useful in cases where other threads may be waiting for
+     * tasks to terminate, or failures must be recorded. Instead consider
+     * using a handler of the form:
+     * <pre> {@code
+     * new RejectedExecutionHandler() {
+     *   public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
+     *     Future<?> dropped = e.getQueue().poll();
+     *     if (dropped != null)
+     *        dropped.cancel(false); // also consider logging the failure
+     *     e.execute(r);             // retry
+     * }}}</pre>
```

Comment:
I tried to reify that code snippet:

```
$ cat MySnippet.java 
import java.util.concurrent.Future;
import java.util.concurrent.RejectedExecutionHandler;
import java.util.concurrent.ThreadPoolExecutor;

public class MySnippet {
    public static void main(String[] args) {
        new RejectedExecutionHandler() {
            public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
                Future<?> dropped = e.getQueue().poll();
                if (dropped != null)
                    dropped.cancel(false); // also consider logging the failure
                e.execute(r);              // retry
            }
        };
    }
}
$ javac MySnippet.java 
MySnippet.java:9: error: incompatible types: Runnable cannot be converted to Future<?>
                Future<?> dropped = e.getQueue().poll();
                                                     ^
1 error
$ 
```
Separately, it seems that the `if` statement uses a 3-space indent which is surprising to see in the JSR 166 code base.